slsReceiverSoftware/src/MasterFileUtility.cpp

```diff
@@ -0,0 +1,358 @@
+// SPDX-License-Identifier: LGPL-3.0-or-other
+// Copyright (C) 2021 Contributors to the SLS Detector Package
+#include "MasterFileUtility.h"
+#include "sls/container_utils.h"
+
+#include <iomanip>
+
+namespace masterFileUtility {
+
+std::string CreateMasterBinaryFile(const std::string filePath,
+                                   const std::string fileNamePrefix,
+                                   const uint64_t fileIndex,
+                                   const bool overWriteEnable,
+                                   const bool silentMode,
+                                   MasterAttributes *attr) {
+    // create file name
+    std::ostringstream os;
+    os << filePath << "/" << fileNamePrefix << "_master"
+       << "_" << fileIndex << ".json";
+    std::string fileName = os.str();
+
+    // create file
+    FILE *fd{nullptr};
+    if (!overWriteEnable) {
+        if (nullptr == (fd = fopen((const char *)fileName.c_str(), "wx"))) {
+            fd = nullptr;
+            throw sls::RuntimeError("Could not create binary master file " +
+                                    fileName);
+        }
+    } else if (nullptr == (fd = fopen((const char *)fileName.c_str(), "w"))) {
+        fd = nullptr;
+        throw sls::RuntimeError(
+            "Could not create/overwrite binary master file " + fileName);
+    }
+
+    rapidjson::StringBuffer s;
+    rapidjson::PrettyWriter<rapidjson::StringBuffer> writer(s);
+    attr->GetBinaryAttributes(&writer);
+    std::string message = s.GetString();
```

Comment:
I think we can save a copy of the string by writing using the char* we get from s.GetString() directly. Length  I think is s.GetLength()

Reply:
s.GetLength() doesnt seem to exist .. googling suggests in older versions..